packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -713,7 +713,11 @@ class RenderIntrinsicHeight extends RenderProxyBox {
 
 }
 
-int _getAlphaFromOpacity(double opacity) => (opacity * 255).round();
+/// Returns an alpha value representative of the provided [opacity] value.
+///
+/// The [opacity] value may not be null, an must be between 0.0 and 1.0,
+/// inclusive.
+int getAlphaFromOpacity(double opacity) => (opacity * 255).round();
```

Comment:
Also clamp opacity to between 0.0 and 1.0, inclusive, and assert that it is non-null.  Just asserting that it is in range is probably not a good idea, since opacity is often animated, and some curves can go negative or greater than 1.0.